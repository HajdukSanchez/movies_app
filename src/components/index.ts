export { MovieList } from './movieList/MovieList';
export { MoviePoster } from './moviePoster/MoviePoster';
export { Loading } from './loading/Loading';
export { MovieDetails } from './movieDetails/MovieDetails';
export { ActorCard } from './actorCard/ActorCard';
