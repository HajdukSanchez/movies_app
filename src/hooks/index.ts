export { useMovies } from './useMovies/useMovies';
export { useMovieDetails } from './useMovieDetails/useMovieDetails';
