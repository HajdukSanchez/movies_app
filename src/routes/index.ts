/**
 * Params for Sack navigation.
 */
export type RootStackParamList = {
  Home: undefined;
  Detail: undefined;
};
