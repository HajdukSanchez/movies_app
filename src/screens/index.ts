export { DetailScreen } from './detailScreen/DetailScreen';
export { HomeScreen } from './homeScreen/HomeScreen';
