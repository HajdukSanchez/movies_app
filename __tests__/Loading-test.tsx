import React from 'react';
import { ActivityIndicator } from 'react-native';
import { render, screen } from '@testing-library/react-native';

import { Loading } from '../src/components';

describe('<Loading/>', () => {
  test('should render correctly', () => {
    const loader = render(<Loading />).toJSON();
    expect(loader).toMatchSnapshot();
  });

  it('should render correctly without props', () => {
    render(<Loading />);
    expect(screen.UNSAFE_getByType(ActivityIndicator)).toBeTruthy();
  });

  it('should render correctly with props', () => {
    render(<Loading size={100} color="red" />);
    expect(screen.UNSAFE_getByType(ActivityIndicator)).toBeTruthy();
    expect(screen.UNSAFE_getByType(ActivityIndicator).props.color).toEqual('red');
    expect(screen.UNSAFE_getByType(ActivityIndicator).props.size).toEqual(100);
  });
});
