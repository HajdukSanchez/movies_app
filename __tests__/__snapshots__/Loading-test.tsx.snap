// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading/> should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    color="red"
    size={100}
  />
</View>
`;
